src/fsharp/service/IncrementalBuild.fs

```diff
@@ -1022,16 +1023,91 @@ module IncrementalBuilderEventTesting =
 
 module Tc = FSharp.Compiler.TypeChecker
 
+[<AutoOpen>]
+module IncrementalBuildSyntaxTree =
 
-/// Accumulated results of type checking.
+    [<Sealed>]
+    type SyntaxTree (tcConfig: TcConfig, fileParsed: Event<string>, lexResourceManager, sourceRange: range, filename: string, isLastCompiland) =
```

Comment:
Should this also be private?

Reply:
I don't think so. We need access to `SyntaxTree` when the enclosing module is auto-opened.

I actually do not want the `IncrementalBuildSyntaxTree` module; instead we should just define `SyntaxTree` without putting in an auto-opened module. The reason why is because `SyntaxTree` is already a module in the compiler, so I can't define it with that name unless I put it in another module, and then open that.